Formula/postgresql@14.rb

```diff
@@ -104,41 +112,94 @@ def post_install
   end
 
   def postgresql_datadir
-    var/"postgres"
+    if old_postgres_data_dir.exist?
+      old_postgres_data_dir
+    else
+      var/name
+    end
   end
 
   def postgresql_log_path
-    var/"log/postgres.log"
+    var/"log/#{name}.log"
   end
 
   def pg_version_exists?
     (postgresql_datadir/"PG_VERSION").exist?
   end
 
+  def old_postgres_data_dir
+    var/"postgres"
+  end
+
+  def postgresql_formula_present?
+    Formula["postgresql"].any_version_installed?
+  end
+
+  # Figure out what version of PostgreSQL the old data dir is
+  # using
+  def old_postgresql_datadir_version
+    pg_version = old_postgres_data_dir/"PG_VERSION"
+    pg_version.exist? && pg_version.read.chomp
+  end
+
   def caveats
-    <<~EOS
-      To migrate existing data from a previous major version of PostgreSQL run:
-        brew postgresql-upgrade-database
+    caveats = ""
+
+    # Extract the version from the formula name
+    pg_formula_version = name.split("@", 2).last
```

Comment:
Any reason we're not just doing
```suggestion
    pg_formula_version = version.to_s
```
here?

Reply:
Shouldn't that be
```suggestion
    pg_formula_version = version.major.to_s
```

Reply:
Yes, that should.

Reply:
Yeh, it's not the full version format and it used to be a different format.